modules/javafx.controls/src/test/java/test/javafx/scene/control/ListCellStartEditTest.java

```diff
@@ -0,0 +1,126 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene.control;
+
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.control.cell.CheckBoxListCell;
+import javafx.scene.control.cell.ChoiceBoxListCell;
+import javafx.scene.control.cell.ComboBoxListCell;
+import javafx.scene.control.cell.TextFieldListCell;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Collection;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Parameterized tests for the {@link ListCell#startEdit()} method of {@link ListCell} and all sub implementations.
+ * The {@link CheckBoxListCell} is special as in there the checkbox will be disabled
+ * based of the editability.
+ */
+@RunWith(Parameterized.class)
+public class ListCellStartEditTest {
+
+    private static final boolean[] EDITABLE_STATES = { true, false };
+
+    private ListView<String> listView;
+    private final ListCell<String> listCell;
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return wrapAsObjectArray(List.of(new ListCell<>(), new ComboBoxListCell<>(), new TextFieldListCell<>(),
+                new ChoiceBoxListCell<>(), new CheckBoxListCell<>(obj -> new SimpleBooleanProperty())));
+    }
+
+    private static Collection<Object[]> wrapAsObjectArray(List<ListCell<?>> listCells) {
+        return listCells.stream().map(tc -> new Object[] { tc }).collect(toList());
+    }
+
+    public ListCellStartEditTest(ListCell<String> listCell) {
+        this.listCell = listCell;
+    }
+
```

Comment:
this hit me when trying separate out the (accidentally?) single test method: all tests work on the same instance of the cell, so its state is unpredictable. Instead, parameterize on a supplier and let it provide a new cell in setup.

Reply:
I see, another point where I would love we actually use JUnit5. :/

Reply:
ehh .. this is not resolved - still change required - or what do I miss?

Reply:
I already changed the tests to use a supplier but didn't pushed it yet. I just resolved this to mark it as read for myself. Will push it today.:)